pkgs/development/compilers/dotnet/build-dotnet.nix

```diff
@@ -121,6 +122,21 @@ stdenv.mkDerivation (finalAttrs: rec {
       "aarch64-darwin" = "osx-arm64";
     };
 
+    updateScript =
+      if type != "sdk" then
+        throw "Only the SDK package can be updated!"
+      else
+      let
+        majorVersion =
+          with lib;
+          concatStringsSep "." (take 2 (splitVersion version));
+      in
+      writeScript "update-dotnet-${majorVersion}" ''
```

Comment:
I think we should use `writeShellScript` since this runs some bash, that sets the shebang for us.